### PR TITLE
362 add rabbitmq to docker

### PR DESCRIPTION
**Pull request by JackVCurtis**
This PR:
 - Updates `docker-compose.yml` to add the RabbitMQ service and config
 - Updates `docket-compose.yml` to use an external volume for MongoDB
 - Adds the local volume mount `.:./src` to the node app so that changes to the code will cause the service to restart
 - Updates `Docker_SETUP.md` and the sample `.env` files for new Docker setup, including data import and export

### CODE REVIEW

File: Docker_SETUP.md [hunk 1]
```diff
@@ -12,14 +12,48 @@ Download Docker for your os type [Docker Website](https://www.docker.com/)
...
+
+## For Onboarding: Export Data
+
+Spin up a container with the volume. Use Ubuntu because we gotta keep Mongo from writing to it.
+
+```sh
+ docker run -v dpdash-mongodb:/dpdash-mongodb --name mongo_backup ubuntu /bin/bash
+```
+
+Copy the volume contents to `./backup.tar`
```
Comment (Elkrival) [inline Docker_SETUP.md:53]: Why do we need a to pack and unpack a backup.tar?

File: docker-compose.yml [hunk 2]
```diff
@@ -4,19 +4,49 @@ services:
...
+      test: ["rabbitmq-diagnostics", "-q", "ping"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
+    environment:
+      - RABBITMQ_NODE_NAME=rabbit@app-tier
+    networks:
+      - app-tier
+networks:
+  app-tier:
```
Comment (Elkrival) [inline docker-compose.yml:47]: Why do we need to add a network..?

**Review by Elkrival — APPROVED** [review]
It's looking good, just a few questions

File: Docker_SETUP.md (hunk 1 above)
Comment (JackVCurtis) [reply Docker_SETUP.md:53]: This is in order to backup the volume containing the MongoDB data. It replaces the db database dump that you gave me in favor of duplicating the Docker volume.

File: docker-compose.yml (hunk 2 above)
Comment (JackVCurtis) [reply docker-compose.yml:47]: This is so RabbitMQ can have a consistent node name/host name. Otherwise Docker will assign something random.

**Pull request merged**